### DISCUSSION TITLE
Some backups don't compress

### DISCUSSION THREAD

**Discussion by tommls** [olahallengren/sql-server-maintenance-solution#188] [Questions & Answers]
Some backups don't compress, can anyone suggest why?? (not all backups, just some, is there anything in these scripts to encourage/ensure compression)

**Comment by olahallengren** [comment]
Could it be that you have TDE on the database? If the database has TDE you need to be on SQL Server 2016, and also specify MAXTRANSFERSIZE for compression to work.
https://blogs.msdn.microsoft.com/sqlcat/2016/06/20/sqlsweet16-episode-1-backup-compression-for-tde-enabled-databases/

Another reason could be if the database contains data that already is compressed (e.g. images).

**Comment by olahallengren** [comment]
Please submit a separate issue for the second question.

**Comment by tommls** [comment]
No TDE, all SQL Server Standard, no images stored in the databases.

**Comment by tommls** [comment]
Separate issue #189 submitted as requested.

**Comment by olahallengren** [comment]
Could you post the backup command from the output file. Does it include WITH COMPRESSION?

**Comment by tommls** [comment]
Where do you store the output file, please??

Once I know, I'll look for it.

The checkbox for compression is ticked in the relevant properties window.
The account running the backups has full access to the databases, backups, etc. etc.

**Comment by olahallengren** [comment]
The only thing that you need to do is to add ```@Compress = 'Y'``` to the backup jobs.

**Comment by olahallengren** [comment]
If you go into Job Step Properties and Advanced, then you can see the location of the output - files.

**Comment by tommls** [comment]
?OIC!! I will do this tonight!! :)

Thank you for helping me!! :)

**Comment by tommls** [comment]
I've done this and compression now works properly.

I also always tick the 'enable compression' box in database properties.
